### PR TITLE
Enable Ruff PLE (Pylint Error)

### PR DESCRIPTION
**Pull request by Avasam**
Ref https://github.com/python/typeshed/issues/13295
https://docs.astral.sh/ruff/rules/#error-e_1
This did found seemingly incorrect dunders in stubs that couldn't be caught by stubtest

### CODE REVIEW

File: stubs/regex/regex/regex.pyi [hunk 1]
```diff
@@ -647,6 +647,6 @@ class Match(Generic[AnyStr]):
     @overload
     def __getitem__(self, key: int | str, /) -> AnyStr | Any: ...
     def __copy__(self) -> Self: ...
-    def __deepcopy__(self) -> Self: ...
+    def __deepcopy__(self, /, memo: Unused) -> Self: ...
```
Comment (AlexWaygood) [inline stubs/regex/regex/regex.pyi:650]: it seems like the `memo` argument is also positional-only:

```
 % uv run --with=regex --no-project python 
Installed 1 package in 5ms
Python 3.13.0 (main, Oct  8 2024, 11:56:29) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import regex
>>> m = regex.match("foo", "foo")
>>> m
<regex.Match object; span=(0, 3), match='foo'>
>>> m.__deepcopy__(memo=None)
Traceback (most recent call last):
  File "<python-input-8>", line 1, in <module>
    m.__deepcopy__(memo=None)
    ~~~~~~~~~~~~~~^^^^^^^^^^^
TypeError: Match.__deepcopy__() takes no keyword arguments
>>> m.__deepcopy__(None)
<regex.Match object; span=(0, 3), match='foo'>
```

File: stubs/regex/regex/regex.pyi [hunk 2]
```diff
@@ -568,7 +568,7 @@ class Pattern(Generic[AnyStr]):
         timeout: float | None = None,
     ) -> _regex.Scanner[bytes]: ...
     def __copy__(self) -> Self: ...
-    def __deepcopy__(self) -> Self: ...
+    def __deepcopy__(self, /, memo: Unused) -> Self: ...
```
Comment (AlexWaygood) [inline stubs/regex/regex/regex.pyi:571]: same here, `memo` appears to be positional-only:

```pycon
>>> regex.compile('foo').__deepcopy__(memo=None)
Traceback (most recent call last):
  File "<python-input-11>", line 1, in <module>
    regex.compile('foo').__deepcopy__(memo=None)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
TypeError: Pattern.__deepcopy__() takes no keyword arguments
>>> regex.compile('foo').__deepcopy__(None)
regex.Regex('foo', flags=regex.V0)
```

File: stubs/regex/regex/regex.pyi (hunk 1 above)
Comment (Avasam) [reply stubs/regex/regex/regex.pyi:650]: I typo'd the location of the /, I meant to make it positional only. Thanks

File: stubs/regex/regex/regex.pyi (hunk 2 above)
Comment (Avasam) [reply stubs/regex/regex/regex.pyi:571]:
```suggestion
    def __deepcopy__(self, memo: Unused, /) -> Self: ...
```

**Comment by AlexWaygood** [comment]
What do we do if we're writing stubs for a library that really does invalidly declare a special method? It's happened before. We'd be in a tough spot there: we'd either have to `noqa` the lint error (due to it ) or allowlist the stubtest error. It might be pretty confusing for a new contributor that they get a CI error whichever thing they do?

**Comment by Avasam** [comment]
> What do we do if we're writing stubs for a library that really does invalidly declare a special method? [...] ]It might be pretty confusing for a new contributor that they get a CI error whichever thing they do?

We've had "unfixable" cases like that, I can namely remember around `__all__` including names that don't exists. I personally think this is one of those rare cases where the best thing to do is to noqa it with a link to a report upstream.

I'd be very surprised, but if we add stubs for a library that does it on purpose, and often. Let's say overloads builtin dunders to add params and explicitly calls those methods instead of using keywords (`del`, `in`, `repr`, etc...). Then we could disable that rule for that stub (similarly to how we disable rules for proto generated stubs)

In other words, I think it's a rare/niche enough case and we have an escape hatch in case it does happen.

**Comment by srittau** [comment]
This has (not suprisingly) conflicts after merging #13306.

**Review by srittau — APPROVED** [review]
Considering this found two genuine problems and did not flag any wrong problems, I support this PR. I also agree with the reasoning of using an explicit `noqa` if code does something unexpected. (And we can also easily drop this check again if it turns out to be annoying.)
